### PR TITLE
[1LP][RFR] IFrame View

### PR DESCRIPTION
**Pull request by izapolsk**

### CODE REVIEW

**Comment by izapolsk** [comment]
one bit is missing. I haven't updated documentation yet. It will be done tomorrow.

**Comment by izapolsk** [comment]
@digitronik , @sbulage, I remember you were interested in this widget. Please look it thru and let me know if you need some extra features.

File: docs/advanced_usage.rst [hunk 1]
```diff
@@ -277,4 +277,37 @@ also make a switchable widget. You can use it like this:
...
 
-Then instead of switching views, it switches widgets.
\ No newline at end of file
+Then instead of switching views, it switches widgets.
+
+IFrame support is views
+-----------------------------
+
+If some html page has embedded iframes, those can be covered using regular view.
+You just need to set FRAME property for it. FRAME should point out to appropriate iframe and can be xpath and whatever supported by widgetastic.
```
Comment (mshriver) [inline docs/advanced_usage.rst]: I think this should be more specific - can it be a webelement? any kind of locator text?

File: docs/advanced_usage.rst [hunk 2]
```diff
@@ -277,4 +277,37 @@ also make a switchable widget. You can use it like this:
...
+
+Since iframe is another page, all its bits consider iframe as root. This has to be taken into account during creating object structure.
+
+If regular views and iframe views are mixed, widgetastic takes care of switching between frames on widget access.
```
Comment (mshriver) [inline docs/advanced_usage.rst]: :tada: :tada: :tada:

File: docs/advanced_usage.rst [hunk 3]
```diff
@@ -277,4 +277,37 @@ also make a switchable widget. You can use it like this:
...
+Since iframe is another page, all its bits consider iframe as root. This has to be taken into account during creating object structure.
+
+If regular views and iframe views are mixed, widgetastic takes care of switching between frames on widget access. 
+User doesn't need to undertake any actions.
+
+Below is example of usage. More examples can be found in unit tests.
+
+.. code-block:: python
+
+    class FirstIFrameView(View):
```
Comment (mshriver) [inline docs/advanced_usage.rst]: Maybe start your example with RegularView, that has FRAME in one of its embedded views.

Consider an example usage call as well, assuming I have a view object for RegularView, can I just call `view.FirstIFrameView.select1.read()`?

File: docs/advanced_usage.rst (hunk 1 above)
Comment (izapolsk) [reply docs/advanced_usage.rst]: of course, it takes everything what is accepted by browser.element()

File: src/widgetastic/widget/base.py [hunk 4]
```diff
@@ -1015,6 +1016,19 @@ def after_fill(self, was_change):
         """
         pass
 
+    def child_widget_accessed(self, widget):
+        """Called when a child widget of this widget gets accessed.
```
Comment (mshriver) [inline src/widgetastic/widget/base.py]: `child widget of this view` <- I think its good to be more explicit here, even though View inherits Widget.

File: docs/advanced_usage.rst (hunk 3 above)
Comment (mshriver) [reply docs/advanced_usage.rst]: I see now in the tests how you're creating two view objects, one for the iFrame view and one for the non-framed view. Both of these are part of the same browser page, and it follows pretty clearly that one could be nested in the other.

**Comment by sbulage** [comment]
LGTM :)

**Pull request merged**